verification/certifyKey_test.go

```diff
@@ -44,18 +100,260 @@ func TestCertifyKey_SimulationMode(t *testing.T) {
+}
+
+// A tcg-dice-Ueid extension MUST be added
+// This SHALL be populated by the LABEL input parameter to CertifyKey
+// The extension SHOULD be marked as critical
+func checkCertifyKeyTcgUeidExtension(t *testing.T, c *x509.Certificate) {
+	t.Helper()
+
+	isFound := false
+	// Check UEID extension
+	for _, ext := range c.Extensions {
+		if ext.Id.String() == ExtensionTcgDiceUeid {
+			isFound = true
+			if !ext.Critical {
+				t.Fatalf("tcg-dice-Ueid extension is NOT marked as CRITICAL")
+			}
+			tcgDiceUEIDString := string(ext.Value)
+			stringSlice := make([]string, len(certifyKeyReq.Label))
+			for i, v := range certifyKeyReq.Label {
+				stringSlice[i] = fmt.Sprintf("%v", v)
+			}
+			certifyKeyReqString := strings.Join(stringSlice, ", ")
+			if tcgDiceUEIDString != certifyKeyReqString {
+				// TODO: Fail when it is exected to pass
+				t.Logf("tcg-dice-Ueid value does not match with the \"Label\" passed in CertifyKeyRequest")
+			}
+			break
+		}
+	}
+	if !isFound {
+		t.Fatalf("tcg-dice-Ueid extension is missing")
+	}
+}
+
+// A tcg-dice-MultiTcbInfo extension.
+// This extension SHOULD be marked as critical.
+func checkCertifyKeyMultiTcbInfoExtension(t *testing.T, c *x509.Certificate) {
+	t.Helper()
+
+	// Check MultiTcbInfo Extension
+	//tcg-dice-MultiTcbInfo extension
+	var multiTcbInfo MultiTcbInfo
+	for _, ext := range c.Extensions {
+		if ext.Id.String() == ExtensionTcgDiceMultiTcbInfo { // OID for Tcg Dice MultiTcbInfo
+			if !ext.Critical {
+				t.Fatalf("TCG DICE MultiTcbInfo extension is not marked as CRITICAL")
+			}
+			_, err := asn1.Unmarshal(ext.Value, &multiTcbInfo)
+			if err != nil {
+				// TODO: Fail when expected to pass
+				t.Logf("Failed to unmarshal MultiTcbInfo field: %v", err)
+			}
+			break
+		}
+	}
+}
+
+// Check whether certificate extended key usage is as per spec
+// OID for ExtendedKeyUsage Extension: 2.5.29.37
+// The ExtendedKeyUsage extension SHOULD be marked as critical
+// If IsCA = true, the extension SHOULD contain tcg-dice-kp-eca
+// If IsCA = false, the extension SHOULD contain tcg-dice-kp-attestLoc
+func checkCertifyKeyExtendedKeyUsages(t *testing.T, c *x509.Certificate) {
+	t.Helper()
+
+	extKeyUsage := []asn1.ObjectIdentifier{}
+
+	for _, ext := range c.Extensions {
+		if ext.Id.String() == ExtensionExtKeyUsage { // OID for ExtKeyUsage extension
+			// Extract the OID value from the extension
+			_, err := asn1.Unmarshal(ext.Value, &extKeyUsage)
+			if err != nil {
+				t.Fatalf("Failed to unmarshal the Extended Key Usage extension: %v", err)
+			}
+
+			if !ext.Critical {
+				t.Fatalf("The Extended Key Usage extension IS NOT CRITICAL, MUST BE CRITICAL")
+			} else {
+				t.Logf("The Extended Key Usage extension is marked CRITICAL")
+			}
+			break
+		}
+	}
+
+	if len(extKeyUsage) == 0 {
+		t.Fatalf("The Extended Key Usage extension is empty")
+	}
+
+	// Iterate over the OIDs in the ExtKeyUsage extension
+	isExtendedKeyUsageValid := false
+	if c.IsCA {
```

Comment:
Only difference between these branches is which OID to look for

Reply:
Done